distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2120,28 +2030,18 @@ private void verifyEvents(
     boolean isAllEventsReceived = false;
     boolean isUnexpectedEventReceived = false;
 
-    while (!isAllEventsReceived && !isUnexpectedEventReceived && millis < MINUTES.toMillis(2)) {
-
-      Set<String> foundIds;
-
-      try {
-        sleep(EVENT_UPDATE_WAIT_INTERVAL);
-        millis += EVENT_UPDATE_WAIT_INTERVAL;
-      } catch (InterruptedException e) {
-        LOGGER.info("Interrupted exception while trying to sleep for events", e);
-      }
-      if ((millis % 1000) == 0) {
-        LOGGER.info("Waiting for events to be received...{}ms", millis);
-      }
-      for (String id : subscriptionIds) {
-        foundIds = getEvents(id);
-        isAllEventsReceived = foundIds.containsAll(metacardIdsExpected);
-
-        isUnexpectedEventReceived = foundIds.removeAll(metacardIdsNotExpected);
-      }
-    }
-    assertTrue(isAllEventsReceived);
-    assertFalse(isUnexpectedEventReceived);
+    await("Waiting for events to be received")
+        .atMost(2, MINUTES)
+        .pollDelay(EVENT_UPDATE_WAIT_INTERVAL, TimeUnit.MILLISECONDS)
+        .until(
+            () ->
```

Comment:
✏️ Move this lambda to a private method to increase readability and avoid nested lambdas